pkg/storagev2/inmemchunkstore/inmemchunkstore.go

```diff
@@ -13,49 +14,110 @@ import (
 )
 
 type ChunkStore struct {
-	chunks sync.Map
+	chunks map[string][]swarm.Chunk
```

Comment:
So we dont need to store the Chunk every time. We only need the Stamp field as this is the only thing that changes. This would save a lot of memory as well.

We should have a separate map for stamps. We can use the same map for the chunks and then also add an entry to this additional map for stamp. The key would be the chunk address itself. Check the implementation in localstorev2/internal/chunkstore pkg for some inspiration.